internal/handlers/tigris/msg_find.go

```diff
@@ -16,11 +16,130 @@ package tigris
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // MsgFind implements HandlerInterface.
 func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	return nil, errNotImplemented
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	unimplementedFields := []string{
+		"skip",
+		"returnKey",
+		"showRecordId",
+		"tailable",
+		"oplogReplay",
+		"noCursorTimeout",
+		"awaitData",
+		"allowPartialResults",
+		"collation",
+		"allowDiskUse",
+		"let",
+	}
+	if err := common.Unimplemented(document, unimplementedFields...); err != nil {
+		return nil, err
+	}
+	ignoredFields := []string{
+		"hint",
+		"batchSize",
+		"singleBatch",
+		"maxTimeMS",
+		"readConcern",
+		"max",
+		"min",
+	}
+	common.Ignored(document, h.L, ignoredFields...)
+
+	var filter, sort, projection *types.Document
+	if filter, err = common.GetOptionalParam(document, "filter", filter); err != nil {
+		return nil, err
+	}
+	if sort, err = common.GetOptionalParam(document, "sort", sort); err != nil {
+		return nil, common.NewErrorMsg(common.ErrTypeMismatch, "Expected field sort to be of type object")
+	}
+	if projection, err = common.GetOptionalParam(document, "projection", projection); err != nil {
+		return nil, err
+	}
+
+	var limit int64
+	if l, _ := document.Get("limit"); l != nil {
```

Comment:
I guess there will be a check somewhere else for `limit` to be valid (like not 0 etc)